x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterLookupJoinIT.java

```diff
@@ -0,0 +1,467 @@
+
+        setSkipUnavailable(REMOTE_CLUSTER_1, true);
+        setSkipUnavailable(REMOTE_CLUSTER_2, false);
+
+        // FIXME: inconsistent with the previous test, remote1:values_lookup still missing, but now it succeeds with remote1 skipped
+        try (
+            EsqlQueryResponse resp = runQuery(
+                "FROM *:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key",
+                randomBoolean()
+            )
+        ) {
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(10));
+            EsqlExecutionInfo executionInfo = resp.getExecutionInfo();
+            assertThat(executionInfo.getClusters().size(), equalTo(2));
+
+            var remoteCluster1 = executionInfo.getCluster(REMOTE_CLUSTER_1);
+            assertThat(remoteCluster1.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SKIPPED));
+            assertThat(remoteCluster1.getFailures(), not(empty()));
+            var failure = remoteCluster1.getFailures().get(0);
+            assertThat(failure.reason(), containsString("lookup index [values_lookup] is not available in remote cluster [cluster-a]"));
+            var remoteCluster2 = executionInfo.getCluster(REMOTE_CLUSTER_2);
+            assertThat(remoteCluster2.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SUCCESSFUL));
+        }
+    }
+
+    public void testLookupJoinMissingLocalIndex() throws IOException {
+        setupClusters(2);
+        populateLookupIndex(REMOTE_CLUSTER_1, "values_lookup", 10);
+
+        var ex = expectThrows(
+            VerificationException.class,
+            () -> runQuery("FROM logs-*,c*:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key", randomBoolean())
+        );
+        assertThat(ex.getMessage(), containsString("lookup index [values_lookup] is not available in local cluster"));
+
+        // Without local in the query it's ok
+        try (
+            EsqlQueryResponse resp = runQuery(
+                "FROM c*:logs-* | EVAL lookup_key = v | LOOKUP JOIN values_lookup ON lookup_key",
+                randomBoolean()
+            )
+        ) {
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(10));
+            var columns = resp.columns().stream().map(ColumnInfoImpl::name).toList();
+            assertThat(columns, hasItems("lookup_key", "lookup_name", "lookup_tag", "v", "tag", "remote_tag"));
+            assertThat(columns, not(hasItems("local_tag")));
+
+            EsqlExecutionInfo executionInfo = resp.getExecutionInfo();
+            assertThat(executionInfo.getClusters().size(), equalTo(1));
+
+            var remoteCluster = executionInfo.getCluster(REMOTE_CLUSTER_1);
+            assertThat(remoteCluster.getStatus(), equalTo(EsqlExecutionInfo.Cluster.Status.SUCCESSFUL));
+        }
+    }
+
+    public void testLookupJoinMissingKey() throws IOException {
```

Comment:
I think this only tests the case when the lookup key is missing on a remote. Maybe let's add the case when it's missing on the local cluster?